src/main/java/org/dmfs/optional/iterator/PresentValues.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.optional.iterator;
+
+import org.dmfs.iterators.AbstractBaseIterator;
+import org.dmfs.iterators.Filter;
+import org.dmfs.iterators.Function;
+import org.dmfs.iterators.decorators.Fluent;
+import org.dmfs.optional.Optional;
+
+import java.util.Iterator;
+
+
+/**
+ * {@link Iterator<E>} that iterates over the present values from the input {@link Iterator<Optional<E>>}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PresentValues<E> extends AbstractBaseIterator<E>
+{
+    private final Iterator<E> mDelegate;
+
+
+    public PresentValues(Iterator<Optional<E>> optionals)
+    {
+        mDelegate = new Fluent<>(optionals)
```

Comment:
On a second thought, I'd like to see this modified a bit. Instead of creating a filtered and mapped delegate, I think in this case it's actually a bit nicer to just create a filtered delegate and do the call to `value()` in `next()`, i.e. only delegate the filtering.
The delegate would become an `Iterator<Optional<E>>` then.
There is not much benefit in delegating the "mapping" as well in this case and the resulting code looks leaner IMO.